packages/create-sitecore-jss/src/templates/nextjs-personalize/src/components/CdpIntegrationScript.tsx

```diff
@@ -21,9 +21,11 @@ interface BoxeverViewEventArgs {
 }
 
 function createPageView(locale: string, routeName: string) {
+  // POS can be multi-valued (one entry per locale) or single valued so we parse it
   // POS must be valid in order to save events (domain name might be taken but it must be defined in CDP settings)
-  const pointOfSale =
-    process.env.NEXT_PUBLIC_CDP_POINTOFSALE || window.location.host.replace(/^www\./, '');
+  // Later on point of sale should be available from layout service data - using env for now
+  const parsedPos = parsePointOfSaleRawInput(process.env.NEXT_PUBLIC_CDP_POINTOFSALE || window.location.host.replace(/^www\./, ''));
+  const pointOfSale = parsedPos[locale] ?? parsedPos['default'];
```

Comment:
Seems a bit strange that the caller has to know about 'default'. Can we combine this logic into one? e.g.

```
const pointOfSale = getPointOfSale(process.env.NEXT_PUBLIC_CDP_POINTOFSALE, locale);
```

We could use this in middleware as well.

Reply:
I agree, it came out of trying to unify the approach for single and multi-value somehow. It would be confusing for helper method to return either Record or string (and assume the consumer will figure out when it's which), so I attempted to always have a Record.
Also in the end we want to have a single value for event (or other things), but we may want to store the multi-value in Record form for reusability. I hope my current implementation would be a good middleground for now and later, when we have more input and usage for point of sale.